dask_kubernetes/core.py

```diff
@@ -154,25 +344,69 @@ def __init__(
         port=None,
         env=None,
         auth=ClusterAuth.DEFAULT,
+        scheduler_timeout=None,
+        local_scheduler=None,
+        interface=None,
+        protocol=None,
+        dashboard_address=None,
+        security=None,
         **kwargs
     ):
-        name = name or dask.config.get("kubernetes.name")
-        namespace = namespace or dask.config.get("kubernetes.namespace")
-        n_workers = (
-            n_workers
-            if n_workers is not None
+        self.pod_template = pod_template
+        self._generate_name = name
+        self._namespace = namespace
+        self._n_workers = n_workers
+        self._scheduler_timeout = scheduler_timeout
+        self._local_scheduler = local_scheduler
+        self._protocol = protocol
+        self._interface = interface
+        self._dashboard_address = dashboard_address
+        self._security = security
+        self.host = host
+        self.port = port
+        self.env = env
+        self.auth = auth
+        self.kwargs = kwargs
+        super().__init__(**self.kwargs)
+
+    async def _start(self):
+        self._generate_name = self._generate_name or dask.config.get("kubernetes.name")
+        self._namespace = self._namespace or dask.config.get("kubernetes.namespace")
+        self._scheduler_timeout = self._scheduler_timeout or dask.config.get(
+            "kubernetes.scheduler-timeout"
+        )
+        self._local_scheduler = (
+            self._local_scheduler
+            if self._local_scheduler is not None
+            else dask.config.get("kubernetes.local-scheduler")
+        )
+        self._n_workers = (
+            self._n_workers
+            if self._n_workers is not None
             else dask.config.get("kubernetes.count.start")
         )
-        host = host or dask.config.get("kubernetes.host")
-        port = port if port is not None else dask.config.get("kubernetes.port")
-        env = env if env is not None else dask.config.get("kubernetes.env")
+        self.host = self.host or dask.config.get("kubernetes.host")
+        self.port = (
+            self.port if self.port is not None else dask.config.get("kubernetes.port")
+        )
+        self._protocol = self._protocol or dask.config.get("kubernetes.protocol")
+        self._interface = self._interface or dask.config.get("kubernetes.interface")
+        self._dashboard_address = self._dashboard_address or dask.config.get(
+            "kubernetes.dashboard_address"
+        )
+        self._security = self._security or dask.config.get("kubernetes.security")
```

Comment:
I think that we might want to verify that this obect is a [`distributed.Security`](https://github.com/dask/distributed/blob/master/distributed/security.py) object.  Although @jcrist would know more about conventions here.

Reply:
Ah ok, so I guess grabbing this from the config might not make sense in this case?

Reply:
You might store a full security object spec as a dictionary under `kubernetes.security`, but I'd guess this would most often be a runtime parameter. I'd leave it out of config for now.

Reply:
Ok thanks! Should I be type checking here too?

Reply:
Probably? Should either be `None` or a `distributed.security.Security` object.